fedimint-server/src/consensus/api.rs

```diff
@@ -377,6 +381,52 @@ impl ConsensusApi {
             .collect()
             .await
     }
+
+    /// Add an API URL announcement from a peer to our database to be returned
+    /// by [`ConsensusApi::api_announcements`].
+    async fn submit_api_announcement(
+        &self,
+        peer_id: PeerId,
+        announcement: SignedApiAnnouncement,
+    ) -> Result<(), ApiError> {
+        if self.cfg.local.identity == peer_id {
+            return Err(ApiError::bad_request(
+                "Cannot submit own announcement".into(),
+            ));
+        }
```

Comment:
I would suggest removing this as its a little unexpected and seems like you yourself forgot when submitting announcements.

Reply:
Agreed